cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,391 @@
+        return ModelUtils.createExecProbe(Arrays.asList("/opt/jmx/jmxtrans_readiness_check.sh", internalBootstrapServiceName, metricsPortValue), kafkaJmxMetricsReadinessProbe);
+    }
+    /**
+     * Wrapper class used to create the overall config file
+     * Servers: A list of servers and what they will query and output
+     */
+    @SuppressWarnings("SE_BAD_FIELD_INNER_CLASS")
+    static class Servers implements Serializable {
+        private static final long serialVersionUID = 1L;
+        public List<Server> servers;
+    }
+
+    /**
+     * Wrapper class used to create the per broker JmxTrans queries and which remote hosts it will output to
+     * host: references which broker the JmxTrans will read from
+     * port: port of the host
+     * queries: what queries are passed in
+     */
+    @SuppressWarnings("SE_BAD_FIELD_INNER_CLASS")
+    static class Server implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        public String host;
+
+        public int port;
+
+        @JsonProperty("queries")
+        public List<Queries> queriesTemplate;
+
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+        public String username;
+
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+        public String password;
+
+    }
+
+    /**
+     * Wrapper class used to create the JmxTrans queries and which remote hosts to output the results to
+     * obj: references what Kafka MBean to reference
+     * attr: an array of what MBean properties to target
+     * outputDefinitionTemplates: Which hosts and how to output to the hosts
+     */
+    @SuppressWarnings("SE_BAD_FIELD_INNER_CLASS")
+    static class Queries implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        public String obj;
+
+        public List<String> attr;
+
+        @JsonProperty("outputWriters")
+        public List<OutputWriter> outputDefinitionTemplates = null;
+    }
+
+    /**
+     * Wrapper class used specify which remote host to output to and in what format to push it in
+     * atClasses: the format of the data to push to remote host
+     * host: The host of the remote host to push to
+     * port: The port of the remote host to push to
+     * flushDelayInSeconds: how often to push the data in seconds
+     */
+    @SuppressWarnings("SE_BAD_FIELD_INNER_CLASS")
+    static class OutputWriter implements Serializable {
+        private static final long serialVersionUID = 1L;
+        @JsonProperty("@class")
+        public String atClasses;
+
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+        public String host;
+
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+        public int port;
+
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+        public int flushDelayInSeconds;
+
+        @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+        public List<String> typeNames;
+    }
```

Comment:
Can we make these toplevel classes, with getters and setters? It would also be nice to have a test that this mapping generates the expected JSON.